platforms/android/src/jp/wizcorp/phonegap/plugin/WizAssets/WizAssetsPlugin.java

```diff
@@ -317,80 +330,147 @@ public AsyncDownload(String url, String uri, String filePath, CallbackContext ca
-                return null;
-            }
+        protected Void doInBackground(File... params) {
+            try {
+                File file = new File(this.filePath);
+
+                // Run async download task
+                File dir = file.getParentFile();
+                if (dir != null && !(dir.mkdirs() || dir.isDirectory())) {
+                    Log.e("WizAssetsPlugin", "Error : subdirectory " + dir.getPath() + " could not be created");
+                    this.callbackContext.error(createDownloadFileError(DIRECTORY_CREATION_ERROR));
+                    return null;
+                }
 
-            Log.d(TAG, "[Downloading] " + file.getAbsolutePath());
+                Log.d(TAG, "[Downloading] " + file.getAbsolutePath());
 
-            try {
-                URL url = new URL(this.url);
-                HttpGet httpRequest = null;
-                httpRequest = new HttpGet(url.toURI());
+                HttpGet httpRequest;
+                try {
+                    URL url = new URL(this.url);
+                    httpRequest = new HttpGet(url.toURI());
+
+                    // Credential check
+                    String credentials = url.getUserInfo();
+                    if (credentials != null) {
+                        // Add Basic Authentication header
+                        httpRequest.setHeader("Authorization", "Basic " + Base64.encodeToString(credentials.getBytes(), Base64.NO_WRAP));
+                    }
+                } catch (MalformedURLException e) {
+                    this.callbackContext.error(createDownloadFileError(INVALID_URL_ERROR));
+                    return null;
+                } catch (URISyntaxException e) {
+                    this.callbackContext.error(createDownloadFileError(INVALID_URL_ERROR));
+                    return null;
+                }
 
+                HttpResponse response;
                 HttpClient httpclient = new DefaultHttpClient();
-
-                // Credential check
-                String credentials = url.getUserInfo();
-                if (credentials != null) {
-                    // Add Basic Authentication header
-                    httpRequest.setHeader("Authorization", "Basic " + Base64.encodeToString(credentials.getBytes(), Base64.NO_WRAP));
+                try {
+	                response = httpclient.execute(httpRequest);
+	                if (response == null) {
+                        this.callbackContext.error(createDownloadFileError(CONNECTIVITY_ERROR));
+                        return null;
+	                }
+	                StatusLine statusLine = response.getStatusLine();
+	                if (statusLine == null) {
+                        this.callbackContext.error(createDownloadFileError(HTTP_REQUEST_ERROR, "No status available"));
+                        return null;
+                    }
+                    int statusCode = statusLine.getStatusCode();
+                    if (statusCode < 200 || statusCode > 299) {
```

Comment:
I doesn't seem to handle redirection by default. I'll double check that and add this param if necessary.
Note: ```DefaultHttpClient``` is currently deprecated, it will have to be replaced later on.